src/test/java/com/google/step/youtube/YouTubeClientMocks.java

```diff
@@ -0,0 +1,149 @@
+    public static final String PROMOCODE = "PROMOCODE";
+    public static final String PROMOCODE_DESCRIPTION = "Get 20% off your first monthly box and enter the code "
+            + PROMOCODE;
+    public static final String DESCRIPTION = "DESCRIPTION WITH NO PROMOCODE";
+    public static final String KEYWORD_DESCRIPTION = "Get 25% off your first purchase at " + KEYWORD + " with the code "
+            + PROMOCODE;
+
+    YouTubeClientMocks() throws IOException {
+        mockYouTubeClient = mock(YouTube.class);
+
+        Channels mockChannels = mock(YouTube.Channels.class);
+        mockListChannels = mock(YouTube.Channels.List.class);
+        when(mockYouTubeClient.channels()).thenReturn(mockChannels);
+        when(mockChannels.list("contentDetails")).thenReturn(mockListChannels);
+        when(mockListChannels.setId(anyString())).thenReturn(mockListChannels);
+        when(mockListChannels.setForUsername(anyString())).thenReturn(mockListChannels);
+
+        PlaylistItems mockPlaylistItems = mock(YouTube.PlaylistItems.class);
+        mockListPlaylistItems = mock(YouTube.PlaylistItems.List.class);
+        when(mockYouTubeClient.playlistItems()).thenReturn(mockPlaylistItems);
+        when(mockPlaylistItems.list("snippet")).thenReturn(mockListPlaylistItems);
+        when(mockListPlaylistItems.setMaxResults(YouTubeInfoScraper.MAX_PLAYLIST_RESULTS))
+                .thenReturn(mockListPlaylistItems);
+        when(mockListPlaylistItems.setPlaylistId(anyString())).thenReturn(mockListPlaylistItems);
+
+        Videos mockVideos = mock(YouTube.Videos.class);
+        mockListVideos = mock(Videos.List.class);
+        when(mockYouTubeClient.videos()).thenReturn(mockVideos);
+        when(mockVideos.list("snippet")).thenReturn(mockListVideos);
+        when(mockListVideos.setId(anyString())).thenReturn(mockListVideos);
+        when(mockListVideos.setFields("items(id, snippet(publishedAt, title, description))"))
+                .thenReturn(mockListVideos);
+
+        Search mockSearch = mock(Search.class);
+        mockListSearch = mock(Search.List.class);
+        when(mockYouTubeClient.search()).thenReturn(mockSearch);
+        when(mockSearch.list("snippet")).thenReturn(mockListSearch);
+        when(mockListSearch.setMaxResults(YouTubeInfoScraper.MAX_SEARCH_RESULTS)).thenReturn(mockListSearch);
+        when(mockListSearch.setQ(anyString())).thenReturn(mockListSearch);
+    }
+
+    public YouTube getYouTubeMock() {
+        return mockYouTubeClient;
+    }
+
+    public Channels.List getMockListChannels() {
+        return mockListChannels;
+    }
+
+    public PlaylistItems.List getMockListPlaylistItems() {
+        return mockListPlaylistItems;
+    }
+
+    public Videos.List getMockListVideos() {
+        return mockListVideos;
+    }
+
+    public Search.List getMockListSearch() {
+        return mockListSearch;
+    }
+
+    public static PromoCode newPromoCode(String description) {
```

Comment:
maybe move the static factory methods into YouTubeClientMocks and rename this class to YouTubeClientMockBuilder.  Utility classes tend are usually noninstantiable and completely stateless (not even immutable non-static state)